tests/test_parallel.py

```diff
@@ -202,3 +234,21 @@ def test_save_results(tmpdir):
         for result_path in result_paths:
             assert lfc.exists(result_path)
             assert Path("local", result_path).exists()
+
+
+def test_iterate_completed_futures():
+    iterations = 5
+    fut = DummyFuture(iterations)
+    start = time.time()
+    fut.result()
+    # If we call result directly, will take ~2 seconds
+    assert time.time() - start > 1
+
+    futures = [DummyFuture(iterations) for _ in range(iterations)]
```

Comment:
Rather than a time-based test, would it be simpler and more robust to test this with mock futures? Thinking of something like:

```suggestion
    futures = [Mock(done=Mock(side_effect=[False] * n + [True])) for n in range(iterations)]
```

Reply:
Will do

Reply:
Removed time dependence in [6527eaa](https://github.com/proteneer/timemachine/pull/1136/commits/6527eaacb30017b935f5b611716383b08bd65b65) doesn't use Mock though as while that can assert the iteration, I wanted to verify that `result` is 'unsafe' to call without the iterating and seems cleaner to define a new class than to create construct an equivalent mock object.